src/Microsoft.Health.Fhir.Core/Extensions/SearchServiceExtensions.cs

```diff
@@ -21,15 +21,20 @@ public static class SearchServiceExtensions
         /// "If-Exists" headers on the API. Additional logic is used to filter parameters that don't restrict
         /// results, and also ensure that the query meets criteria requirements
         /// </summary>
-        public static async Task<IReadOnlyCollection<SearchResultEntry>> ConditionalSearchAsync(this ISearchService searchService, string instanceType, IReadOnlyList<Tuple<string, string>> conditionalParameters, CancellationToken cancellationToken)
+        public static async Task<(IReadOnlyCollection<SearchResultEntry> results, string continuationToken)> ConditionalSearchAsync(
```

Comment:
>ConditionalSearchAsync [](http://example.com/codeflow?start=109&length=22)

It's getting overly complicated...

I would break it into two method:
`ConditionalSearchForSingleEntity`
and `ConditionalSearchForCollection`.

Only change this method is needed is to actually force _count=2 (no need to search for more than two entities)

For second method maybe it make sense to pass token and count as separate typed parameters... instead of List of tuples...
Or maybe don't have second method at all, and do all that code in handler itself.

I just don't like how for one specific case we have to alter logic in 3 different places unrelated to it.